.meta/sinks/honeycomb.toml

```diff
@@ -0,0 +1,42 @@
+[sinks.honeycomb]
+title = "Honeycomb"
+beta = true
+common = false
+delivery_guarantee = "best_effort"
+egress_method = "batching"
+function_category = "transmit"
+healthcheck = true
+input_types = ["log"]
+write_to_description = "[Honeycomb][urls.honeycomb] via the [batch events API][urls.honeycomb_batch]"
+requirements = {}
+
+<%= render("_partials/_component_options.toml", type: "sink", name: "honeycomb") %>
+
+<%= render("_partials/_batch_options.toml", namespace: "sinks.honeycomb.options", common: false, max_events: nil, max_size: 5242880, timeout_secs: 1) %>
+
+<%= render("_partials/_buffer_options.toml", namespace: "sinks.honeycomb.options") %>
+
+<%= render(
+  "_partials/_request_options.toml",
+  namespace: "sinks.honeycomb.options",
+  common: false,
+  in_flight_limit: 5,
+  rate_limit_duration_secs: 1,
+  rate_limit_num: 5,
+  retry_attempts: -1,
+  retry_initial_backoff_secs: 1,
+  retry_max_duration_secs: 10,
+  timeout_secs: 60
+) %>
+
+[sinks.honeycomb.options.api_key]
+type = "string"
+examples = ["${MY_API_KEY}", "some-api-key"]
+required = true
+description = "The team key that will be used to authenticate against Honeycomb."
+
+[sinks.honeycomb.options.dataset]
+type = "string"
+examples = ["my-honeycomb-dataset"]
```

Comment:
Can we make this field templateable?

Reply:
From the PR description

> There is one blocker, I'd like to change dataset to be template-able but we require a valid dataset for the healthcheck to work. But due to the fact that healthchecks do not get any events we need a default. This also is an issue for running it regularly, we need some way to force a default onto templates since it is not possible to send events without a valid dataset.

Reply:
Yeah, I read that. A few comments:

1. I think Honeycomb creates datasets on the fly. Ex: if an even specified a new data set it is created.
2. If a user supplies a variable the healthcheck should be disabled, just like it does on other sinks with this behavior.
3. If the variable results in a blank value then the request should be dropped, just like it would for any non-retryable error.

I don't think any of these things should block implementation, but I am also ok shipping this in it's current form.

Reply:
Sounds good :+1: I like that

Reply:
I've opened an issue for this and will add this feature in a follow up PR https://github.com/timberio/vector/issues/2049